workflows/prognostic_c48_run/runtime/overrider.py

```diff
@@ -0,0 +1,130 @@
+import dataclasses
+import functools
+import logging
+from typing import Hashable, Mapping, MutableMapping, Set
+
+import cftime
+import fsspec
+import xarray as xr
+
+import fv3gfs.util
+from runtime.monitor import Monitor
+from runtime.types import Diagnostics, Step
+from runtime.derived_state import DerivedFV3State
+from runtime.utils import quantity_state_to_ds, ds_to_quantity_state
+
+QuantityState = MutableMapping[Hashable, fv3gfs.util.Quantity]
+
+logger = logging.getLogger(__name__)
+
+
+@dataclasses.dataclass
+class OverriderConfig:
+    """Configuration for overriding tendencies from a step.
+    
+    Attributes:
+        url: path to dataset containing tendencies.
+        variables: mapping from state name to name of corresponding tendency in
+            provided dataset. For example: {"air_temperature": "fine_res_Q1"}.
+    """
+
+    url: str
+    variables: Mapping[str, str]
+
+
+@dataclasses.dataclass
+class OverriderAdapter:
+    """Wrap a Step function to override certain tendencies."""
+
+    config: OverriderConfig
+    state: DerivedFV3State
```

Comment:
Suggest against using the hard-coded `DerivedFV3State` class name which can only be made online in favor of the general purpose `runtime.types.State`. This is what I did for the emulator. Technically your tests below would not pass the type-checker.

Reply:
Using `runtime.types.State` leads to:
```
workflows/prognostic_c48_run/runtime/overrider.py:69: error: "MutableMapping[Hashable, DataArray]" has no attribute "time"```

Reply:
Ah. you need the time attribute. ok...